src/System.Numerics.Vectors/src/System/Numerics/Vector.tt

```diff
@@ -468,49 +467,151 @@ namespace System.Numerics
         /// <returns>The hash code.</returns>
         public override int GetHashCode()
         {
-            int hash = 0;
-
             if (Vector.IsHardwareAccelerated)
```

Comment:
Is this specialized code a problem?

Reply:
In general, yes.  The rest of the .tt code uses the `supportedTypes` list, but here we are duplicating that list.  If we ever wanted to add a new supported type, this would need to be updated as well.